extensions/common/context.cc

```diff
@@ -68,6 +68,57 @@ void extractServiceName(const std::string& fqdn, std::string* service_name) {
   }
 }
 
+void populateRequestInfo(bool outbound, RequestInfo* request_info) {
+  // TODO: switch to stream_info.requestComplete() to avoid extra compute.
+  request_info->end_timestamp = getCurrentTimeNanoseconds();
+
+  // Fill in request info.
+  // Try to get fqdn of destination service from cluster name. If not found, use
+  // host header instead.
+  std::string cluster_name = "";
+  getStringValue({"cluster_name"}, &cluster_name);
+  extractFqdn(cluster_name, &request_info->destination_service_host);
+  if (!request_info->destination_service_host.empty()) {
+    // cluster name follows Istio convention, so extract out service name.
+    extractServiceName(request_info->destination_service_host,
```

Comment:
Can you merge master? I feel like I just saw a variant of this by @bianpengyuan . Can we combine the two?

Reply:
Looks like the pr is not merged yet... https://github.com/istio/proxy/pull/2548...

Reply:
Merged master and updated PR